### ISSUE TITLE
building with sdl fails on openbsd 6.6 (immintrin.h no such file or directory)

### ISSUE THREAD

**Issue by reitraced** [simh/simh#822]
- #### Context

i was trying to build the vax simulator on openbsd 6.6 with the current github master branch
- #### the output of "sim> SHOW VERSION" while running the simulator which is having the issue
latest on github, cant build it to tell which exact version
- #### how you built the simulator or that you're using prebuilt binaries

gmake with gcc
- #### the simulator configuration file (or commands) which were used when the problem occurred.

n/a
- #### the expected behavior and the actual behavior

```
In file included from /usr/local/include/SDL2/SDL.h:38,
                 from ./sim_video.h:229,
                 from ./VAX/vax_vc.c:41:
/usr/local/include/SDL2/SDL_cpuinfo.h:86:23: error: immintrin.h: No such file or directory
In file included from /usr/local/include/SDL2/SDL.h:38,
                 from ./sim_video.h:229,
                 from ./VAX/vax_lk.h:33,
                 from ./VAX/vax_lk.c:37:
/usr/local/include/SDL2/SDL_cpuinfo.h:86:23: error: immintrin.h: No such file or directory
In file included from /usr/local/include/SDL2/SDL.h:38,
                 from ./sim_video.h:229,
                 from ./VAX/vax_vs.h:33,
                 from ./VAX/vax_vs.c:38:
/usr/local/include/SDL2/SDL_cpuinfo.h:86:23: error: immintrin.h: No such file or directory
In file included from /usr/local/include/SDL2/SDL.h:38,
                 from ./sim_video.h:229,
                 from ./scp.c:228:
/usr/local/include/SDL2/SDL_cpuinfo.h:86:23: error: immintrin.h: No such file or directory
./sim_tape.c: In function 'sim_tape_tpc_map':
./sim_tape.c:3280: warning: comparison is always false due to limited range of data type
In file included from /usr/local/include/SDL2/SDL.h:38,
                 from ./sim_video.h:229,
                 from ./sim_video.c:30:
/usr/local/include/SDL2/SDL_cpuinfo.h:86:23: error: immintrin.h: No such file or directory
gmake: *** [makefile:2242: BIN/vax] Error 1
```
- #### you may also need to provide specific pointers to data files that may be necessary to demonstrate the problem

**Comment by reitraced** [comment]
also a note, this happens on the pdp11 sim as well so i think its all that use sdl

**Comment by markpizz** [comment]
This sounds like a defect in the SDL2 development packaging on your system.

Simh doesn't do anything special here but attempt to use what it finds available on the host.  If you find a way to get it to work let us know.

Meanwhile, you can build without SDL if you uninstall the SDL2 development package on your host system, OR you rename the usr/include/SDL2 directory to SDL2.xxx so it isn't found by the makefile.

**Comment by markpizz** [comment]
My earlier comment was done from my phone where I couldn't see all the details.  I didn't see that you were working with SDL in /usr/local.  

The issue template asked you to supply all of the output that was produced when building the simulator(s).  I would have had more to say if you had supplied those details.

In any case, as I said earlier, either fix the broken SDL2 setup or avoid using it.

I suspect that you've built the SDL2 package on your own.  That is specifically not a supported configuration.  You built it, you own the problems.  If your OS has a packaged (and tested) libSDL2 and it's related development components, those would be supported.

Good Luck.

**Comment by reitraced** [comment]
its uhh
the one that came with openbsd. it puts all user installed stuff in /usr/local by default
even if its the binaries from them

**Comment by reitraced** [comment]
like i used pkg_add to get it, and it put it there

**Comment by reitraced** [comment]
and heres the full output, sorry for not putting that earlier. this time on pdp11
```
$ gmake pdp11
lib paths are: /usr/lib /usr/X11R6/lib /usr/local/lib
include paths are:  /usr/include /usr/local/include
using libm: /usr/lib/libm.a
using libpthread: /usr/lib/libpthread.a /usr/include/pthread.h
using libpcre: /usr/local/lib/libpcre.a /usr/local/include/pcre.h
using semaphore: /usr/include/semaphore.h
using libdl: /usr/include/dlfcn.h
using libpng: /usr/local/lib/libpng.a /usr/local/include/png.h
using zlib: /usr/lib/libz.a /usr/include/zlib.h
using mman: /usr/include/sys/mman.h
using libSDL2: /usr/local/include/SDL2/SDL.h
using libpcap: /usr/include/pcap.h
***
*** pdp11 Simulator being built with:
*** - compiler optimizations and no debugging support. GCC Version: 4.2.1.
*** - dynamic networking support using OpenBSD provided libpcap components.
*** - Local LAN packet transports: PCAP TAP NAT(SLiRP)
*** - video capabilities provided by libSDL2 (Simple Directmedia Layer).
*** - Per simulator tests will be run.
***
*** git commit id is 45d7a1155b642c0af21ac2d254ae59114d95e14f.
*** git commit time is 2020-03-18T14:13:55+0100.
***
gcc -std=gnu99 -U__STRICT_ANSI__  -O2 -finline-functions -fgcse-after-reload -fno-unsafe-loop-optimizations -fno-strict-overflow -DSIM_GIT_COMMIT_ID=45d7a1155b642c0af21ac2d254ae59114d95e14f -DSIM_GIT_COMMIT_TIME=2020-03-18T14:13:55+0100  -DSIM_COMPILER="GCC Version: 4.2.1" -DSIM_BUILD_TOOL=simh-makefile -I . -D_GNU_SOURCE -I/usr/local/include -DUSE_READER_THREAD -DSIM_ASYNCH_IO  -DHAVE_PCRE_H -DHAVE_SEMAPHORE -DHAVE_SYS_IOCTL -DHAVE_DLOPEN=so -DHAVE_UTIME -DHAVE_LIBPNG -DHAVE_ZLIB -DHAVE_GLOB -DHAVE_SHM_OPEN  ./PDP11/pdp11_fp.c ./PDP11/pdp11_cpu.c ./PDP11/pdp11_dz.c ./PDP11/pdp11_cis.c ./PDP11/pdp11_lp.c ./PDP11/pdp11_rk.c ./PDP11/pdp11_rl.c ./PDP11/pdp11_rp.c ./PDP11/pdp11_rx.c ./PDP11/pdp11_stddev.c ./PDP11/pdp11_sys.c ./PDP11/pdp11_tc.c ./PDP11/pdp11_tm.c ./PDP11/pdp11_ts.c ./PDP11/pdp11_io.c ./PDP11/pdp11_rq.c ./PDP11/pdp11_tq.c ./PDP11/pdp11_pclk.c ./PDP11/pdp11_ry.c ./PDP11/pdp11_pt.c ./PDP11/pdp11_hk.c ./PDP11/pdp11_xq.c ./PDP11/pdp11_xu.c ./PDP11/pdp11_vh.c ./PDP11/pdp11_rh.c ./PDP11/pdp11_tu.c ./PDP11/pdp11_cpumod.c ./PDP11/pdp11_cr.c ./PDP11/pdp11_rf.c ./PDP11/pdp11_dl.c ./PDP11/pdp11_ta.c ./PDP11/pdp11_rc.c ./PDP11/pdp11_kg.c ./PDP11/pdp11_ke.c ./PDP11/pdp11_dc.c ./PDP11/pdp11_dmc.c ./PDP11/pdp11_kmc.c ./PDP11/pdp11_dup.c ./PDP11/pdp11_rs.c ./PDP11/pdp11_vt.c ./PDP11/pdp11_td.c ./PDP11/pdp11_io_lib.c ./PDP11/pdp11_rom.c ./PDP11/pdp11_ch.c ./display/display.c ./display/sim_ws.c ./display/vt11.c ./PDP11/pdp11_ng.c ./PDP11/pdp11_daz.c ./display/ng.c ./scp.c ./sim_console.c ./sim_fio.c ./sim_timer.c ./sim_sock.c ./sim_tmxr.c ./sim_ether.c ./sim_tape.c ./sim_disk.c ./sim_serial.c ./sim_video.c ./sim_imd.c ./sim_card.c -DVM_PDP11 -I ./PDP11 -DHAVE_PCAP_NETWORK -I/usr/include/ -DBPF_CONST_STRING -DUSE_SHARED -DHAVE_TAP_NETWORK -DHAVE_BSDTUNTAP -Islirp -Islirp_glue -Islirp_glue/qemu -DHAVE_SLIRP_NETWORK -DUSE_SIMH_SLIRP_DEBUG slirp/*.c slirp_glue/*.c -DUSE_DISPLAY -DHAVE_LIBSDL -DUSE_SIM_VIDEO `/usr/local/bin/sdl2-config --cflags` `/usr/local/bin/sdl2-config --libs` -o BIN/pdp11 -L/usr/lib -L/usr/X11R6/lib -L/usr/local/lib -lm -lpthread -lpcre -L/usr/local/lib/ -lpng -lz
In file included from /usr/local/include/SDL2/SDL.h:38,
                 from ./sim_video.h:229,
                 from ./PDP11/pdp11_vt.c:50:
/usr/local/include/SDL2/SDL_cpuinfo.h:86:23: error: immintrin.h: No such file or directory
In file included from /usr/local/include/SDL2/SDL.h:38,
                 from ./sim_video.h:229,
                 from ./display/sim_ws.c:41:
/usr/local/include/SDL2/SDL_cpuinfo.h:86:23: error: immintrin.h: No such file or directory
In file included from /usr/local/include/SDL2/SDL.h:38,
                 from ./sim_video.h:229,
                 from ./PDP11/pdp11_ng.c:31:
/usr/local/include/SDL2/SDL_cpuinfo.h:86:23: error: immintrin.h: No such file or directory
In file included from /usr/local/include/SDL2/SDL.h:38,
                 from ./sim_video.h:229,
                 from ./PDP11/pdp11_daz.c:29:
/usr/local/include/SDL2/SDL_cpuinfo.h:86:23: error: immintrin.h: No such file or directory
In file included from /usr/local/include/SDL2/SDL.h:38,
                 from ./sim_video.h:229,
                 from ./scp.c:228:
/usr/local/include/SDL2/SDL_cpuinfo.h:86:23: error: immintrin.h: No such file or directory
./sim_tape.c: In function 'sim_tape_tpc_map':
./sim_tape.c:3280: warning: comparison is always false due to limited range of data type
In file included from /usr/local/include/SDL2/SDL.h:38,
                 from ./sim_video.h:229,
                 from ./sim_video.c:30:
/usr/local/include/SDL2/SDL_cpuinfo.h:86:23: error: immintrin.h: No such file or directory
gmake: *** [makefile:2222: BIN/pdp11] Error 1
```

**Comment by markpizz** [comment]
You might want to raise the issue with the folks who maintain packages for your OS.

Note that the only include file simh looks for and actually includes is SDL.h.  Anything else is included within the SDL/OS provided components, not all of which are there or correctly setup.

Since SDL is broken, I suggest that you either remove the SDL development package (which you're probably not using directly since it seems that references to it are broken), or merely make it not findable by renaming the /usr/local/include/SDL2 directory to something else.

**Comment by reitraced** [comment]
yeah ive been renaming it
what does sdl even do in simh anyways?

**Comment by markpizz** [comment]
Your problem might be that libSDL2's development package could incorrectly describe other packages that it depends on.  You may be able to find the right combination of packages with some effort.  This, again, would be a problem to raise with the package maintainers.

**Comment by markpizz** [comment]
Some simulated systems have devices which provide simulations of various video/displays that were options for those original systems.  Most of those graphics interfaces were quite expensive, :-) These days, not counting your system, the experience is much cheaper.

The PDP11 has several graphics devices.

The various VAX simulators have graphics support which then made the systems known as "VAXStations".

The PDP10-KA and PDP10-KI had some graphics interfaces which are now emulated.

**Comment by markpizz** [comment]
Closing this issue now.

**Comment by vtamara** [comment]
I had the same issue in OpenBSD/adJ 6.8.

However I notice the problem is not with SDL but with the compiler.

```
pkglocate immintrin.h
comp68:/usr/lib/clang/10.0.1/include/immintrin.h
gcc-8.3.0p5:lang/gcc/8,-main:/usr/local/lib/gcc/x86_64-unknown-openbsd6.7/8.3.0/include/immintrin.h
llvm-8.0.1p11:devel/llvm,-main:/usr/local/lib/clang/8.0.1/include/immintrin.h
```

As you see, immintrin.h is provided by clang 10 of the base system and could be provided by the package gcc 8.3 or the package llvm-8.  However it is not provided  by the default gcc  4.2.1 included with OpenBSD 6.8.

At least simh v4.0-Beta-1 tries to use gcc by default, and then it will compile with gcc 4.2.1 that doesn't provide immintrin.h.  

I have had not success compiling with clang, but I could compile with gcc 8.3 (whose main binary is egcc).  To do that:

doas pkg_add gcc
GCC=egcc gmake 



immintrin.h is provided by clang included
